### PR TITLE
Adding top operations to execution plan 

### PR DESCRIPTION
**Pull request by aasimkhan30**

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Localization/sr.strings [hunk 1]
```diff
@@ -2234,6 +2234,22 @@ WarningOverlayTooltip = Warnings
...
+EstimatedCost = Estimated Cost %
+EstimatedSubtree = Estimated Subtree Cost %
+ActualRows = Actual Rows
+EstimatedRows = Estimated Rows
+ActualExecutions = Actual Executions
+EstimatedExecutions = Estimated Executions
+EstimatedCpu = Estimated CPU Cost
+EstimatedIO = Estimated IO Cost
+ActualDataSize = Actual Data Size
+EstiamtedDataSize = Est Data Size
```
Comment (lewis-sanchez) [inline src/Microsoft.SqlTools.ServiceLayer/Localization/sr.strings:2251]: Misspelling for `EstimatedDataSize`

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs [hunk 2]
```diff
@@ -163,6 +164,173 @@ public static List<ExecutionPlanGraphPropertyBase> GetProperties(PropertyDescrip
...
+                });
+            }
+
+            if (currentNode["EstimateCPU"] != null)
+            {
+                result.Add(new TopOperationsDataItem
+                {
+                    ColumnName = SR.EstimatedCpu,
+                    DataType = PropertyValueDataType.Number,
+                    DisplayValue = currentNode["EstimateCPU"]
```
Comment (lewis-sanchez) [inline src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs:247]: Same as above

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs [hunk 3]
```diff
@@ -163,6 +164,173 @@ public static List<ExecutionPlanGraphPropertyBase> GetProperties(PropertyDescrip
...
+                });
+            }
+
+            if (currentNode["EstimateIO"] != null)
+            {
+                result.Add(new TopOperationsDataItem
+                {
+                    ColumnName = SR.EstimatedIO,
+                    DataType = PropertyValueDataType.Number,
+                    DisplayValue = currentNode["EstimateIO"]
```
Comment (lewis-sanchez) [inline src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs:257]: Same as aboce

File: src/Microsoft.SqlTools.ServiceLayer/Localization/sr.cs [hunk 4]
```diff
@@ -8485,6 +8485,102 @@ public static string ParallelismOverlayTooltip
...
+
+        public static string ActualDataSize
+        {
+            get
+            {
+                return Keys.GetString(Keys.ActualDataSize);
+            }
+        }
+
+        public static string EstiamtedDataSize
```
Comment (lewis-sanchez) [inline src/Microsoft.SqlTools.ServiceLayer/Localization/sr.cs:8576]: Rename to "EstimatedDataSize"

File: src/Microsoft.SqlTools.ServiceLayer/Localization/sr.cs [hunk 5]
```diff
@@ -12950,6 +13046,42 @@ public class Keys
...
+            public const string EstimatedCpu = "EstimatedCpu";
+
+
+            public const string EstimatedIO = "EstimatedIO";
+
+
+            public const string ActualDataSize = "ActualDataSize";
+
+
+            public const string EstiamtedDataSize = "EstiamtedDataSize";
```
Comment (lewis-sanchez) [inline src/Microsoft.SqlTools.ServiceLayer/Localization/sr.cs:13082]: Correct property spelling and value spelling to "EstimatedDataSize"

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs (hunk 3 above)
Comment (aasimkhan30) [reply src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs:257]: Unfortunately, these properties names are coming from the engine and there's not much we can do on our end. If you read a sqlplan file you should see these properties named like this.

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ShowPlan/Node.cs [hunk 6]
```diff
@@ -595,7 +590,8 @@ public string[] GetDisplayLinesOfText(bool roundCostForSmallGraph = false)
 
             if (!this.HasPDWCost || cost > 0)
             {
-                if(roundCostForSmallGraph && this.graph != null && this.graph.NodeStmtMap.Count < 20){
+                if (roundCostForSmallGraph && this.graph != null && this.graph.NodeStmtMap.Count < 20)
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ShowPlan/Node.cs:593]: Why 20? It'd be good to make this into a constant and then add a comment explaining where that value comes from so it's clear what it is and how it's being used.

File: src/Microsoft.SqlTools.ServiceLayer/Localization/sr.resx [hunk 7]
```diff
@@ -4644,6 +4644,54 @@ The Query Processor estimates that implementing the following index could improv
...
+  <data name="EstimatedIO" xml:space="preserve">
+    <value>Estimated IO Cost</value>
+    <comment></comment>
+  </data>
+  <data name="ActualDataSize" xml:space="preserve">
+    <value>Actual Data Size</value>
+    <comment></comment>
+  </data>
+  <data name="EstiamtedDataSize" xml:space="preserve">
+    <value>Est Data Size</value>
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Localization/sr.resx:4692]: Why is this shortened but all the other ones fully spell out Estimated?

**Review by Charles-Gagnon — APPROVED** [review]

**Review by lewis-sanchez — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/Localization/sr.resx (hunk 7 above)
Comment (aasimkhan30) [reply src/Microsoft.SqlTools.ServiceLayer/Localization/sr.resx]: Fixed it.

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ShowPlan/Node.cs (hunk 6 above)
Comment (aasimkhan30) [reply src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ShowPlan/Node.cs:593]: For larger plans, we wanted to show the cost in decimals. For that, we have decided the threshold as 20 for now.

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs [hunk 8]
```diff
@@ -163,6 +164,173 @@ public static List<ExecutionPlanGraphPropertyBase> GetProperties(PropertyDescrip
...
+            {
+                result.Add(new TopOperationsDataItem
+                {
+                    ColumnName = SR.EstimatedRebinds,
+                    DataType = PropertyValueDataType.Number,
+                    DisplayValue = currentNode["EstimateRebinds"]
+                });
+            }
+
+            if (currentNode["Partitioned"] != null)
```
Comment (alanrenmsft) [inline src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanGraphUtils.cs:322]: please define consts for these strings

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ShowPlan/Node.cs (hunk 6 above)
Comment (alanrenmsft) [reply src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ShowPlan/Node.cs:593]: pls add comment in the code about why 20 is picked.

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/Contracts/ExecutionPlanGraph.cs [hunk 9]
```diff
@@ -206,4 +210,11 @@ public enum PropertyValueDataType
         Boolean = 2,
         Nested = 3
     }
+
+    public class TopOperationsDataItem
+    {
+        public string ColumnName { get; set; }
+        public PropertyValueDataType DataType { get; set; }
+        public object DisplayValue { get; set; }
```
Comment (alanrenmsft) [inline src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/Contracts/ExecutionPlanGraph.cs:218]: display value should always be string, right?

**Review by alanrenmsft — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ShowPlan/Node.cs (hunk 6 above)
Comment (Charles-Gagnon) [reply src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ShowPlan/Node.cs:593]: And use a const for magic values like this - it makes it easier to understand if you have a variable with a clear name

**Pull request merged**